prime_server/prime_server.hpp

```diff
@@ -169,7 +169,7 @@ class worker_t {
   };
   using interrupt_function_t = std::function<void()>;
   using work_function_t =
-      std::function<result_t(const std::list<zmq::message_t>&, void*, interrupt_function_t&)>;
+      std::function<result_t(const std::list<zmq::message_t>&, void*, const interrupt_function_t&)>;
```

Comment:
this is a breaking API change and isnt needed for this pr, im considering reverting it and the others like it